src/webpack-plugins/generate-json.ts

```diff
@@ -3,24 +3,35 @@ import buildBundles from '../scripts/build-bundles';
 import generateComponentsJSON from '../utils/generate-components-json';
 import start from '../server/';
 
-export default class GenerateJSON {
 
+export default class GenerateJSON {
     private shouldGenerateJSON: boolean = false;
 
-    apply(compiler) {
-        compiler.plugin('invalid', () => {
-            this.shouldGenerateJSON = true;
-        });
+    fullReload = (_, cb) => {
+        if (this.shouldGenerateJSON) {
```

Comment:
I think it's better to use `this.shouldGenerateJSON === true` for a better readability